src/tests/FastBoard_unittest.cpp

```diff
@@ -0,0 +1,342 @@
+    
+    // Negative test to check assertion
+    // Commenting out until assertions are enable in CI build.
+    //ASSERT_DEATH({ b.get_xy(7);}, ".*FastBoard.cpp.* Assertion `y >= 0 && y < m_boardsize' failed.");
+}
+
+TEST(FastBoardTest, GetSquare) {
+    FastBoard b;
+    b.reset_board(19);
+    EXPECT_EQ(FastBoard::EMPTY, b.get_square(43));
+    EXPECT_EQ(FastBoard::EMPTY, b.get_square(0, 1));
+    b.set_square(43, FastBoard::BLACK);
+    EXPECT_EQ(FastBoard::BLACK, b.get_square(43));
+    b.set_square(43, FastBoard::WHITE);
+    EXPECT_EQ(FastBoard::WHITE, b.get_square(43));
+}
+
+TEST(FastBoardTest, SemiFilled5x5Board) {
+    FastBoard b = create_filled_5x5();
+    
+    const char *expected = "\n"
+        "   a b c d e \n"
+        " 5 . . O . .  5\n"
+        " 4 X . O . .  4\n"
+        " 3 . . O X .  3\n"
+        " 2 . X X O .  2\n"
+        " 1 . . . . .  1\n"
+        "   a b c d e \n\n";
+    
+    EXPECT_EQ(expected,  b.serialize_board());
+}
+
+// Results will make more sense in FullBuard test
+TEST(FastBoardTest, CountRealLibertiesOn5x5) {
+    FastBoard b = create_filled_5x5();
+    EXPECT_EQ(2, b.count_pliberties(b.get_vertex(0, 0)));
+    EXPECT_EQ(4, b.count_pliberties(b.get_vertex(1, 1)));
+    EXPECT_EQ(4, b.count_pliberties(b.get_vertex(2, 1)));
+    EXPECT_EQ(4, b.count_pliberties(b.get_vertex(3, 1)));
+    EXPECT_EQ(3, b.count_pliberties(b.get_vertex(4, 1)));
+    EXPECT_EQ(4, b.count_pliberties(b.get_vertex(2, 2)));
+    EXPECT_EQ(4, b.count_pliberties(b.get_vertex(3, 2)));
+    EXPECT_EQ(3, b.count_pliberties(b.get_vertex(0, 3)));
+}
+
+TEST(FastBoardTest, SemiFilled9x9Board) {
+    FastBoard b = create_filled_9x9();
+    
+    const char *expected = "\n"
+        "   a b c d e f g h j \n"
+        " 9 . . . . . . . . .  9\n"
+        " 8 . . . . . . . . .  8\n"
+        " 7 . . + . + . O . .  7\n"
+        " 6 . . . . O . . . .  6\n"
+        " 5 . . + . + O + . .  5\n"
+        " 4 . . X . O X O . .  4\n"
+        " 3 . X X . + O + . .  3\n"
+        " 2 . . . . . . . . .  2\n"
+        " 1 X . . . . . . . .  1\n"
+        "   a b c d e f g h j \n\n";
+    
+    EXPECT_EQ(expected,  b.serialize_board());
+}
+
+// Results will make more sense in FullBuard test
+TEST(FastBoardTest, CountRealLibertiesOn9x9) {
+    FastBoard b = create_filled_9x9();
+    
+    EXPECT_EQ(2, b.count_pliberties(b.get_vertex(0, 0)));
+    EXPECT_EQ(4, b.count_pliberties(b.get_vertex(1, 2)));
+    EXPECT_EQ(4, b.count_pliberties(b.get_vertex(4, 3)));
+    EXPECT_EQ(4, b.count_pliberties(b.get_vertex(4, 4))); 
+    EXPECT_EQ(4, b.count_pliberties(b.get_vertex(5, 4)));
+}
+
+// Results will make more sense in FullBuard test
+TEST(FastBoardTest, IsSuicideWhenNotForBlack) {
+    FastBoard b;
+    b.reset_board(5);  
+    b.set_square(2, 2, FastBoard::WHITE);
+    EXPECT_EQ(false, b.is_suicide(b.get_vertex(1, 1), FastBoard::BLACK));
+    EXPECT_EQ(false, b.is_suicide(b.get_vertex(2, 1), FastBoard::BLACK));
```

Comment:
Wouldn't testing the is_suicide=True case be useful?

Reply:
Yes, it would be more usful, but since update_board is not part of FastBoard, there is no way to do it!  is_suicide will always be false unless we can move update_board to FastBoard. Then it could be properly tested here. As it is now, it can only be tested in the context of a FullBoard (which I have done on a separate branch).